API/views.py

```diff
@@ -28,34 +31,50 @@ def get42(url, payload, token):
 	return response.json()
 
 def get_token(CODE):
-	url = 'https://api.intra.42.fr/oauth/token'
-	UID = os.environ.get("UID_APP")
-	SECRET = os.environ.get("SECRET_KEY_APP")
-
-	data = {
-		'grant_type': 'authorization_code',
-		'client_id':  UID,
-		'client_secret': SECRET,
-		'code': CODE,
-		'redirect_uri': 'http://localhost:8000/twofa'
-	}
-
-	response = requests.post(url, data=data)
-	return response.json()['access_token']
+	try:
+		url = 'https://api.intra.42.fr/oauth/token'
+		UID = os.environ.get("UID_APP")
+		SECRET = os.environ.get("SECRET_KEY_APP")
+
+		data = {
+			'grant_type': 'authorization_code',
+			'client_id':  UID,
+			'client_secret': SECRET,
+			'code': CODE,
+			'redirect_uri': 'http://localhost:8000/twofa'
+		}
+
+		response = requests.post(url, data=data)
+	
+		return response.json()['access_token']
+	except:
+		raise Exception("Error get_token") 
 
 def create_user_API(code):
-	token = get_token(code)
-	if Users.objects.filter(token_42=hashlib.md5(token.encode()).hexdigest()).exists():
-		user = Users.objects.get(token_42=hashlib.md5(token.encode()).hexdigest())
+	
+	if Users.objects.filter(code_42=hashlib.md5(code.encode()).hexdigest()).exists():
```

Comment:
so in this func we try to check if a user alrdy exist in the db before we try to create it.
so why there are 3 calls to the db?
one by `code_42` then `username` and then `token_2FA`?